Firmware/ultralcd.cpp

```diff
@@ -7427,3 +7437,52 @@ void lcd_heat_bed_on_load_toggle()
         value = !value;
     eeprom_update_byte((uint8_t*)EEPROM_HEAT_BED_ON_LOAD_FILAMENT, value);
 }
+
+void reprint_from_eeprom() {
+	char cmd[30];
+	char filename[13];
+	char altfilename[13];
+	uint8_t depth = 0;
+	char dir_name[9];
+
+	enableReprint=false;
+
+	//cmdqueue_reset();
+
+	depth = eeprom_read_byte((uint8_t*)EEPROM_DIR_DEPTH);
+
+	MYSERIAL.println(int(depth));
+	for (int i = 0; i < depth; i++) {
+		for (int j = 0; j < 8; j++) {
+			dir_name[j] = eeprom_read_byte((uint8_t*)EEPROM_DIRS + j + 8 * i);
+		}
+		dir_name[8] = '\0';
+		MYSERIAL.println(dir_name);
+		// strcpy(dir_names[i], dir_name);
+		card.chdir(dir_name, false);
+	}
+
+	for (int i = 0; i < 8; i++) {
+		filename[i] = eeprom_read_byte((uint8_t*)EEPROM_FILENAME + i);
+	}
+	filename[8] = '\0';
```

Comment:
We can probably save quite a bit of memory by re-using `restore_file_from_sd()` here from one of my PRs: https://github.com/prusa3d/Prusa-Firmware/pull/4437/files

Reply:
@gudnimg the only lack about your approach is related with the files which have the _.g_ extension instead of _.gco_
That's something that were already discussed on the original PR #3071, here is the [link for the comments](https://github.com/prusa3d/Prusa-Firmware/pull/3071#discussion_r615726244)
Because of that I added this block I know that it was not the more elegant solution, but it works

	strcpy(altfilename,filename);
	if (!card.FileExists(altfilename))
	{
		strcat_P(filename, PSTR(".gco"));
		if (card.FileExists(filename))
		{
			strcpy(altfilename,filename);
		}else
		{
			strcat_P(altfilename, PSTR(".g"));
		}
	}

Reply:
I see 🤔 The power panic code should also handle the `.g` extension but it always assume it is `.gco` (Its been that way for years). I think it is time we add the short file name extension into EEPROM. 

We can’t expand `EEPROM_FILENAME` since it will affect adjacent memory. But we can add the extension in a different location. I can fix it in my PR since your PR is unrelated to power panic functionality. 

I think we can reduce the flash increase of your PR from +456 bytes to something closer to +200 bytes. 

We can probably merge your PR first and then later I can create a separate PR to “optimise” it because the flash memory we have left is precious.

Reply:
That make sense, if you need anything from my side just tell me.

Reply:
Resolving this thread for now.

Reply:
I pushed a fix to my PR to support any valid DOS 8.3 file extension. Only tested `".gco"` and `".g"` on my end and it works. I can confirm the current MK3 branch does not work for `".g"`

This did cost ~100 bytes of flash though, but I think it will be worth it since the Re-print functionality can re-use the code.